### PR TITLE
Update the rust version and depdnecy

### PR DESCRIPTION
**Pull request by larry-aptos**
* Update the dependency version and rust version to be compatible with aptos-core.

### CODE REVIEW

**Comment by larry-aptos** [comment]
* **#515** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-indexer-processors/515?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @larry-aptos and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by larry-aptos** [comment]
* **#515** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-indexer-processors/515?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @larry-aptos and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by larry-aptos** [comment]
* **#515** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-indexer-processors/515?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @larry-aptos and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: rust/Cargo.toml [hunk 1]
```diff
@@ -83,10 +83,11 @@ kanal = { version = "0.1.0-pre8", features = ["async"] }
 once_cell = "1.10.0"
 num_cpus = "1.16.0"
 pbjson = "0.5.1"
-prometheus = { version = "0.13.0", default-features = false }
+prometheus = { version = "0.13.3", default-features = false }
 prost = { version = "0.12.3", features = ["no-recursion-limit"] }
 prost-types = "0.12.3"
-rayon = "1.10.0"
+# Keep it compatible with the aptos-core version.
+rayon = "1.5.2"
```
Comment (just-in-chang) [inline rust/Cargo.toml:90]: Hopefully nothing breaks with this change lol, 1.10 -> 1.5.2 seems like a big change

**Review by just-in-chang — APPROVED** [review]

File: rust/Cargo.toml (hunk 1 above)
Comment (larry-aptos) [reply rust/Cargo.toml:90]: this is the version aptos-core uses and here it's introduced by @grao1991 

let me know if you have any concern. I feel upgrade rayon from 1.5.2 to 1.10 at aptos-core side is more aggressive 🤦

Comment (larry-aptos) [reply rust/Cargo.toml:90]: Going to merge and we can revisit this if there is performance degradation.

**Pull request merged**